### PR TITLE
Nov 2024

### PR DESCRIPTION
**Pull request by SethBarberee**
The great renaming of move/orb actions that haven't been documented yet. Still a few more orb funcs in 807CD9C that I would like to finish before marking this PR ready.

### CODE REVIEW

**Comment by SethBarberee** [comment]
`HandleTrawlOrbAction` is a little too much for with the huge stack it has so left the WIP scratch and matched the rest of the orb funcs. With that, this PR is ready for review.

File: src/move_actions.c [hunk 1]
```diff
@@ -306,7 +306,8 @@ bool8 YawnMoveAction(Entity * pokemon, Entity *target, Move *move, s32 param_4)
     return TRUE;
 }
 
-bool8 sub_80576F8(Entity * pokemon, Entity *target, Move *move, s32 param_4)
+// NOTE: Is there a better name for this?
+bool8 BasicSleepMoveAction(Entity * pokemon, Entity *target, Move *move, s32 param_4)
```
Comment (AnonymousRandomPerson) [inline src/move_actions.c:310]: Perhaps the "Basic" isn't needed. `SleepMoveAction` by itself conveys the same meaning of being a shared move action between similar sleep moves.

File: src/status_actions.c [hunk 2]
```diff
@@ -787,13 +785,13 @@ bool8 CleanseOrbAction(Entity * pokemon, Entity * target, Move *move, s32 param_
     return isItemCleaned;
 }
 
-bool8 sub_805BC70(Entity * pokemon, Entity * target, Move *move, s32 param_4)
+bool8 SiestaMoveAction(Entity * pokemon, Entity * target, Move *move, s32 param_4)
```
Comment (AnonymousRandomPerson) [inline src/status_actions.c:788]: Siesta is the Slumber Orb's "move".

File: src/status_actions.c [hunk 3]
```diff
@@ -678,7 +676,7 @@ bool8 sub_805BA44(Entity * pokemon, Entity * target, Move *move, s32 param_4)
     return TRUE;
 }
 
-bool8 sub_805BA50(Entity * pokemon, Entity * target, Move *move, s32 param_4)
+bool8 TakeawayMoveAction(Entity * pokemon, Entity * target, Move *move, s32 param_4)
```
Comment (AnonymousRandomPerson) [inline src/status_actions.c:679]: Takeaway is used by the Mug Orb.

**Comment by AnonymousRandomPerson** [comment]
Some merge conflicts were introduced with https://github.com/pret/pmd-red/pull/256 merging.

File: src/move_actions.c (hunk 1 above)
Comment (Kermalis) [reply src/move_actions.c:310]: If this is referring to rest etc, then Sleep is fine. If it's referring to forcing sleep (like hypnosis), then I suggest "CastSleep".
Same for "CastFreeze" and "CastBurn" etc if that's the situation

**Review by mateon1 — APPROVED** [review]

File: src/move_actions.c (hunk 1 above)
Comment (SethBarberee) [reply src/move_actions.c:310]: It doesn't "force" the status conditions. It still has to roll the chance for a status condition. For example,
```c
bool8 BasicFireMoveAction(Entity *pokemon, Entity *target, Move *move, u32 param_4)
{
  bool8 flag;

  flag = FALSE;
  SendThawedMessage(pokemon, target);
  if (HandleDamagingMove(pokemon,target,move,0x100,param_4) != 0) {
    flag = TRUE;
    if(sub_805727C(pokemon, target, gUnknown_80F4DBA))
    {
        BurnedStatusTarget(pokemon, target, 0, FALSE);
    }
  }
  return flag;
}

```

File: src/called_move_data.c [hunk 4]
```diff
@@ -2,300 +2,300 @@
...
-extern bool8 sub_8058478(Entity *pokemon, Entity *target, Move *move, s32 param_4);
-extern bool8 sub_80584C0(Entity *pokemon, Entity *target, Move *move, s32 param_4);
-extern bool8 sub_8058548(Entity *pokemon, Entity *target, Move *move, s32 param_4);
+extern bool8 FlatterMoveAction(Entity *pokemon, Entity *target, Move *move, s32 param_4);
+extern bool8 WillOWispMoveAction(Entity *pokemon, Entity *target, Move *move, s32 param_4);
+extern bool8 ReturnMoveAction(Entity *pokemon, Entity *target, Move *move, s32 param_4);
+extern bool8 FlameWheelMoveAction(Entity *pokemon, Entity *target, Move *move, s32 param_4);
+extern bool8 GustMoveAction(Entity *pokemon, Entity *target, Move *move, s32 param_4);
+extern bool8 DisableMoveAction(Entity *pokemon, Entity *target, Move *move, s32 param_4);
+extern bool8 CrunchMoveAction(Entity *pokemon, Entity *target, Move *move, s32 param_4);
```
[the remark this answers is not in the document]
Comment (SethBarberee) [reply src/called_move_data.c:25]: Sure, given it's used more for Shadow Ball, that's fine with me.

File: src/status_actions.c (hunk 3 above)
Comment (SethBarberee) [reply src/status_actions.c:681]: Yes and no. This also gets called for Thief and Covet.

sub_805A2A0 (Thief and Covet) will call this function. Might name sub_805A2A0 `ThiefMoveAction`.

Comment (AnonymousRandomPerson) [reply src/status_actions.c:681]: Ah okay, in that case naming it after Thief would be more recognizable than naming it after Takeaway.

**Comment by AnonymousRandomPerson** [comment]
@SethBarberee FYI, there are some unresolved comments here that I'm waiting on before approving.

**Comment by SethBarberee** [comment]
Was on vacation and just got back. Will try to address them in next day or two.

**Review by AnonymousRandomPerson — APPROVED** [review]

**Pull request merged**